.codegen/service.py.tmpl

```diff
@@ -71,46 +79,31 @@ class {{.PascalName}}{{if eq "List" .PascalName}}Request{{end}}:{{if .Descriptio
 {{- end -}}
 {{- define "type" -}}
 	{{- if not . }}any # ERROR: No Type
-	{{- else if .ArrayValue }}List[{{template "type-nq" .ArrayValue}}]
-	{{- else if .MapValue }}Dict[str,{{template "type-nq" .MapValue}}]
+	{{- else if .ArrayValue }}List[{{template "type" .ArrayValue}}]
+	{{- else if .MapValue }}Dict[str,{{template "type" .MapValue}}]
 	{{- else if .IsExternal }}{{.Package.Name}}.{{.PascalName}}
 	{{- else if .IsObject }}{{.PascalName}}{{if eq "List" .PascalName}}Request{{end}}
 	{{- else if .Enum }}{{.PascalName}}
-	{{- else}}{{template "type-nq" .}}{{- end -}}
-{{- end -}}
-{{- define "type-nq" -}}
-	{{- if .IsString}}str
-	{{- else if .IsEmpty}}Any
+	{{- else if .IsString}}str
 	{{- else if .IsAny}}Any
+	{{- else if .IsEmpty}}{{.PascalName}}
 	{{- else if .IsBool}}bool
 	{{- else if .IsInt64}}int
 	{{- else if .IsFloat64}}float
 	{{- else if .IsInt}}int
 	{{- else if .IsByteStream}}BinaryIO
-	{{- else if .ArrayValue }}List[{{template "type-nq" .ArrayValue}}]
-	{{- else if .MapValue }}Dict[str,{{template "type-nq" .MapValue}}]
-	{{- else if .IsExternal }}{{.Package.Name}}.{{.PascalName}}
-	{{- else if .IsObject }}{{.PascalName}}{{if eq "List" .PascalName}}Request{{end}}
-	{{- else if .Enum }}{{.PascalName}}
 	{{- else}}any /* MISSING TYPE */
 	{{- end -}}
 {{- end -}}
 
 {{- define "type-doc" -}}
-	{{- if .IsString}}str
-	{{- else if .IsEmpty}}Any
-	{{- else if .IsAny}}Any
-	{{- else if .IsBool}}bool
-	{{- else if .IsInt64}}int
-	{{- else if .IsFloat64}}float
-	{{- else if .IsInt}}int
-	{{- else if .IsByteStream}}BinaryIO
+	{{- if .IsEmpty}}:class:`{{template "type" .}}`
 	{{- else if .ArrayValue }}List[{{template "type-doc" .ArrayValue}}]
 	{{- else if .MapValue }}Dict[str,{{template "type-doc" .MapValue}}]
 	{{- else if .IsExternal }}:class:`{{.PascalName}}`
 	{{- else if .IsObject }}:class:`{{.PascalName}}{{if eq "List" .PascalName}}Request{{end}}`
 	{{- else if .Enum }}:class:`{{.PascalName}}`
-	{{- else}}any /* MISSING TYPE */
+	{{- else}}{{template "type" . }}
```

Comment:
This removes unnecessary duplication between type and type-doc, and removes the completely duplicated type-nq template. There is no behavior change here, other than changing `.IsEmpty` from `Any` to `{{.PascalName}}`.